File: src/main/java/frc/team5190/robot/navigation/NAVCommand.java
package frc.team5190.robot.navigation;

import com.ctre.phoenix.motorcontrol.ControlMode;
import edu.wpi.first.wpilibj.command.Command;
import frc.team5190.robot.Robot;

public class NAVCommand extends Command
{
    Arcade.Pos current = new Arcade.Pos(0, 0, 90);
    Arcade.Pos target  = new Arcade.Pos(10, 10, 0);


    public NAVCommand()
    {
        requires(Robot.driveTrain);
        requires(Robot.navigation);
    }

    @Override
    protected void initialize()
    {
        Robot.navigation.init(current, target);
    }

    @Override
    protected void execute()
    {
        Robot.navigation.drive();
    }

    @Override
    protected boolean isFinished()
    {
        return Robot.navigation.isFinished();
    }

    @Override
    protected void end()
    {
        Robot.driveTrain.tankDrive(0, 0, ControlMode.PercentOutput);
    }
}

File: src/main/kotlin/frc/team5190/robot/navigation/NAVCommand.kt
package frc.team5190.robot.navigation

import com.ctre.phoenix.motorcontrol.ControlMode
import edu.wpi.first.wpilibj.command.Command
import edu.wpi.first.wpilibj.smartdashboard.SmartDashboard
import frc.team5190.robot.MainXbox
import frc.team5190.robot.drive.DriveSubsystem

class NAVCommand(private val path: NAVHelper) : Command() {

    init {
        requires(DriveSubsystem)
    }

    private lateinit var leftMotionProfile: NAVFeeder
    private lateinit var rightMotionProfile: NAVFeeder

    var paused = false

    override fun initialize() {
        leftMotionProfile = NAVFeeder(DriveSubsystem.falconDrive.leftMaster, path.trajectoryLeft)
        leftMotionProfile.startMotionProfile()

        rightMotionProfile = NAVFeeder(DriveSubsystem.falconDrive.rightMaster, path.trajectoryRight)
        rightMotionProfile.startMotionProfile()
    }

    override fun execute() {
        SmartDashboard.putBoolean("Obstacle Present", MainXbox.bButton)

        if (MainXbox.bButton) {
            leftMotionProfile.pauseMotionProfile()
            rightMotionProfile.pauseMotionProfile()
            paused = true
            return
        } else if (paused) {
            leftMotionProfile.resumeMotionProfile()
            rightMotionProfile.resumeMotionProfile()
        }

        leftMotionProfile.control()
        rightMotionProfile.control()

        DriveSubsystem.falconDrive.leftMaster.set(ControlMode.MotionProfile, leftMotionProfile.getSetValue().value.toDouble())
        DriveSubsystem.falconDrive.rightMaster.set(ControlMode.MotionProfile, rightMotionProfile.getSetValue().value.toDouble())
    }

    override fun end() {
        leftMotionProfile.reset()
        rightMotionProfile.reset()
        DriveSubsystem.falconDrive.tankDrive(ControlMode.PercentOutput, 0.0, 0.0)
    }

    override fun isFinished() = leftMotionProfile.hasFinished() && rightMotionProfile.hasFinished()
}